### PR TITLE
Add the total sum of minutes and moves

### PR DESCRIPTION
**Pull request by huishi329**

### CODE REVIEW

**Comment by akaka0039** [comment]
sorry one thing
the day which user selected will disappear( i guess the problem is text color)
IMG_7857

File: react-app/src/Mine/index.js [hunk 1]
```diff
@@ -7,14 +7,40 @@ import TodaySummary from './TodaySummary'
 import Setting from '../Setting'
 import WeekCalendar from './WeekCalendar'
 
-export default function Mine({ db, setReset, reset }) {
+export default function Mine({ db, setReset, reset, exercises }) {
+  const [todayRecords, setTodayRecords] = useState([])
+  const [allRecords, setAllRecords] = useState([])
+  const [startDate, setStartDate] = useState('')
+  const toYYYYMMDD = (timestamp) => {
+    new Date(timestamp).toISOString().split('T')[0]
```
Comment (mylesieong) [inline react-app/src/Mine/index.js:15]: Do we need return here?

File: react-app/src/Mine/index.js [hunk 2]
```diff
@@ -7,14 +7,40 @@ import TodaySummary from './TodaySummary'
...
+        'select timestamp, exercise_name from Record where timestamp between ? and ? order by timestamp',
+        [localStartOfDay, localStartOfDay + 24 * 60 * 60 * 1000 - 1],
+        (_, { rows }) => {
+          setTodayRecords(rows._array)
+        }
+      )
+      tx.executeSql('select * from Record', [], (_, { rows }) => {
+        setAllRecords(rows._array)
+      })
+      tx.executeSql('select Min(timestamp) from limit 1', [], (_, { rows }) => {
```
Comment (mylesieong) [inline react-app/src/Mine/index.js:31]: Should it be select Min() from Record Limit 1?

File: react-app/src/Mine/WeekCalendar.js [hunk 3]
```diff
@@ -4,50 +4,66 @@ import { CalendarProvider, WeekCalendar as Week } from 'react-native-calendars'
...
+      }),
+      {}
+    )
+    setCalendarRecords(newCalendarRecords)
+  }, [records])
 
   return (
     <View style={styles.container}>
-      <Text style={styles.info}>Your exercise records since 03/08/2023</Text>
+      <Text style={styles.info}>Your exercise records since {startDate}</Text>
```
Comment (mylesieong) [inline react-app/src/Mine/WeekCalendar.js:57]: Should be You've been exercising since xxx

File: react-app/src/Mine/index.js [hunk 4]
```diff
@@ -7,14 +7,40 @@ import TodaySummary from './TodaySummary'
...
+        setStartDate(toYYYYMMDD(rows._array[0]))
```
Comment (j3rryzh) [inline react-app/src/Mine/index.js:32]: Should be (rows._array[0].timestamp)?

File: react-app/src/Mine/index.js (hunk 1 above)
Comment (j3rryzh) [reply react-app/src/Mine/index.js:15]: Oh yep, I think should add a `return` in front of `new Date…`

File: react-app/src/Mine/WeekCalendar.js [hunk 5]
```diff
@@ -4,50 +4,70 @@ import { CalendarProvider, WeekCalendar as Week } from 'react-native-calendars'
...
+      {}
+    )
+    setCalendarRecords(newCalendarRecords)
+  }, [records])
 
   return (
     <View style={styles.container}>
-      <Text style={styles.info}>Your exercise records since 03/08/2023</Text>
+      {startDate ? (
+        <Text style={styles.info}>Your have been exercising since {startDate}</Text>
```
Comment (j3rryzh) [inline react-app/src/Mine/WeekCalendar.js:58]: Is it `You` instead of `Your`?

**Review by j3rryzh — APPROVED** [review]

**Review by akaka0039 — APPROVED** [review]

**Pull request merged**